x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationQueryParams.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.search;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Query parameters for a {@link org.elasticsearch.xpack.application.search.action.QuerySearchApplicationAction}. For now, they contain
+ * just the template parameters to override in the search request, but can potentially include more options and parameters in the future
+ */
+public class SearchApplicationQueryParams implements ToXContentObject, Writeable {
+
+    public static final ParseField TEMPLATE_PARAMS_FIELD = new ParseField("params");
+    private static final ConstructingObjectParser<SearchApplicationQueryParams, Void> PARSER = new ConstructingObjectParser<>(
+        "query_params",
+        p -> {
+            @SuppressWarnings("unchecked")
+            final Map<String, Object> templateParams = (Map<String, Object>) p[0];
+            return new SearchApplicationQueryParams(templateParams);
+        }
+    );
+
+    static {
+        PARSER.declareObject(optionalConstructorArg(), (p, c) -> p.map(), TEMPLATE_PARAMS_FIELD);
+    }
+
+    private final Map<String, Object> templateParams;
+
+    public SearchApplicationQueryParams() {
+        this(Map.of());
+    }
+
+    public SearchApplicationQueryParams(Map<String, Object> templateParams) {
+        // TODO Understand how can we make the parser build the object using the default constructor instead of this one
```

Comment:
Why is it needed?

Reply:
I changed the parsing code - this is no longer an issue.